code/script.js

```diff
@@ -222,104 +224,172 @@ const generateBoard = () => {
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category].includes(value))
+    } else {
+      alert(
+        `No, this person doesn't have ${value} eyes, removing all people with ${value} eyes!`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => !person[category].includes(value))
+    }
+  } else if(category === 'hair'){
+    if (keep) {
+      alert(
+       `Yes, the mystery person has ${value} hair, keep all people with ${value} hair!`
+      )
+      peopleInPlay = peopleInPlay.filter((person) => person[category].includes(value))
+    } else{
+      alert(
+      `No the person does not have ${value} hair. Removing all with ${value} hair. `
+      )
+      peopleInPlay = peopleInPlay.filter((person) => !person[category].includes(value))
+    }
+      // Invoke a function to redraw the board with the remaining people.
+      generateBoard()
   }
-
+}
   // Determine what is the category
   // filter by category to keep or remove based on the keep variable.
   /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      example 1
+      peopleInPlay = poepleInPlay.filter((person) => person[category] === value)
+      peopleInPlay = poepleInPlay.filter((person) => person[category] !=== value)
+   
+      exampel 2
+      peopleInPlay = peopleInPlay.filter((person) => person[category].includes(value))
+      peopleInPlay = peopleInPlay.filter((person) => !person[category].includes(value))
+      
+      The two examples executes the same thing
   */
 
-  // Invoke a function to redraw the board with the remaining people.
-}
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
+//STEP 8 ----
+//when clicking guess, the player first have to confirm that they want to make a guess.
+ // If the player wants to guess, the checkMyGuess function is invoked.
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+ confirmGuess= confirm(`Do you really want to make this guess?`)
+if (confirmGuess){
+  checkMyGuess(personToConfirm)
 }
+}
+
 
+
+//STEP 9 ----
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
   // 1. Check if the personToCheck is the same as the secret person's name
   // 2. Set a Message to show in the win or lose section accordingly
   // 3. Show the win or lose section
   // 4. Hide the game board
+
+const checkMyGuess = (personToCheck) => {
```

Comment:
This also works perfectly. For having the restart button showing you should not need to target it separately. I think that since it is a part of the whole section in the html and css for win or lose, it should show as a part of that. I remember struggling a bit with this one and ended up with a function just for this win or lose part, within the checkMyGuess function, which I had to call separately. I'm sure there's a much smoother way. :-)